internal/pkg/archiver/walker.go

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package archiver
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// FileItem is unit of work for archive
+type FileItem struct {
+	FullPath string
+	RelPath  string
+	FileInfo os.FileInfo
+	Link     string
+}
+
+// Walker provides a channel of file info/paths for archival
+//
+//nolint: gocyclo
+func Walker(ctx context.Context, rootPath string) (<-chan FileItem, <-chan error, error) {
```

Comment:
This could probably be turned into a more generic library. I think it has some overlap with `osctl ls`. Just something to think about I think for v0.2.

Reply:
yep, good point, I somehow forgot that we have it there as well. I can unify both to use this Walker function which should work well for `ls`/`cp`.